internal/mage/sdk/python.go

```diff
@@ -0,0 +1,107 @@
+package sdk
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"dagger.io/dagger"
+	"github.com/dagger/dagger/internal/mage/util"
+	"github.com/magefile/mage/mg"
+)
+
+const (
+	pythonGeneratedAPIPath = "sdk/python/src/dagger/codegen.py"
+)
+
+var _ SDK = Python{}
+
+type Python mg.Namespace
+
+// Lint lints the Python SDK
+func (t Python) Lint(ctx context.Context) error {
+	c, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	_, err = pythonBase(c).
+		Exec(dagger.ContainerExecOpts{
+			Args: []string{"hatch", "run", "lint:style"},
+		}).
+		ExitCode(ctx)
+	if err != nil {
+		return err
+	}
+
+	return lintGeneratedCode(pythonGeneratedAPIPath, func() error {
+		return t.Generate(ctx)
+	})
+}
+
+// Test tests the Python SDK
+func (t Python) Test(ctx context.Context) error {
+	c, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	_, err = pythonBase(c).
+		Exec(dagger.ContainerExecOpts{
+			Args:                          []string{"hatch", "run", "test"},
+			ExperimentalPrivilegedNesting: true,
+		}).
+		ExitCode(ctx)
+	return err
+}
+
+// Generate re-generates the SDK API
+func (t Python) Generate(ctx context.Context) error {
+	c, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	generated, err := pythonBase(c).
+		Exec(dagger.ContainerExecOpts{
+			Args:                          []string{"hatch", "run", "generate"},
+			ExperimentalPrivilegedNesting: true,
+		}).
+		Exec(dagger.ContainerExecOpts{
+			Args: []string{"hatch", "run", "lint:fmt"},
+		}).
+		File(strings.TrimPrefix(pythonGeneratedAPIPath, "sdk/python/")).
+		Contents(ctx)
+	if err != nil {
+		return err
+	}
+
+	return os.WriteFile(pythonGeneratedAPIPath, []byte(generated), 0600)
+}
+
+func pythonBase(c *dagger.Client) *dagger.Container {
+	src := c.Directory().WithDirectory("/", util.Repository(c).Directory("sdk/python"))
+
+	base := c.Container().From("python:3.10-alpine").
+		Exec(dagger.ContainerExecOpts{
+			Args: []string{"apk", "add", "-U", "--no-cache", "gcc", "musl-dev", "libffi-dev"},
+		})
+
+	base = base.
+		WithEnvVariable("PIP_NO_CACHE_DIR", "off").
+		WithEnvVariable("PIP_DISABLE_PIP_VERSION_CHECK", "on").
+		WithEnvVariable("PIP_DEFAULT_TIMEOUT", "100").
+		Exec(dagger.ContainerExecOpts{
+			Args: []string{"pip", "install", "hatch"},
+		})
+
+	return base.
+		WithWorkdir("/app").
+		WithMountedDirectory("/app", src).
+		Exec(dagger.ContainerExecOpts{
+			Args: []string{"hatch", "env", "create"},
+		})
```

Comment:
You can add another `hatch env create lint` to create the lint env here. Otherwise it'll be created on next `hatch run lint:...`.

Reply:
Dang. I thought I added this already